src/github/api/url.rs

```diff
@@ -0,0 +1,75 @@
+use anyhow::Context as _;
+use reqwest::header::HeaderValue;
+
+/// A URL to a GitHub API endpoint.
+/// When using a GitHub App instead of a PAT, the token depends on the organization.
+/// So storing the token together with the URL is convenient.
+#[derive(Clone)]
+pub struct GitHubUrl {
+    url: String,
+    auth: HeaderValue,
+}
+
+impl GitHubUrl {
+    pub fn new(url: String, auth: HeaderValue) -> Self {
+        let https = "https://";
+        let url = if url.starts_with(https) {
+            url
+        } else {
+            format!("{https}api.github.com/{url}")
+        };
+        Self { url, auth }
+    }
+
+    pub fn new_with_org(url: String, org: &str) -> anyhow::Result<Self> {
+        let auth = get_token_from_env(org).and_then(|token| auth_header(&token))?;
+        Ok(Self::new(url, auth))
+    }
+
+    pub fn repos(org: &str, repo: &str, remaining_endpoint: &str) -> anyhow::Result<Self> {
+        let remaining_endpoint = if remaining_endpoint.is_empty() {
+            "".to_string()
+        } else {
+            validate_remaining_endpoint(remaining_endpoint)?;
+            format!("/{remaining_endpoint}")
+        };
+        let url = format!("repos/{org}/{repo}{remaining_endpoint}");
+        Self::new_with_org(url, org)
+    }
+
+    pub fn orgs(org: &str, remaining_endpoint: &str) -> anyhow::Result<Self> {
+        validate_remaining_endpoint(remaining_endpoint)?;
+        let url = format!("orgs/{org}/{remaining_endpoint}");
+        Self::new_with_org(url, org)
+    }
+
+    pub fn url(&self) -> &str {
+        &self.url
+    }
+
+    pub fn auth(&self) -> &HeaderValue {
+        &self.auth
+    }
+}
+
+fn validate_remaining_endpoint(endpoint: &str) -> anyhow::Result<()> {
+    anyhow::ensure!(
+        !endpoint.starts_with('/'),
+        "remaining endpoint {endpoint} should not start with a slash"
+    );
+    Ok(())
+}
+
+fn get_token_from_env(org: &str) -> anyhow::Result<String> {
+    std::env::var(format!("GITHUB_TOKEN_{}", &org.to_uppercase()))
```

Comment:
This is fine as a quick hack, but otherwise it's ugly :laughing: We should parse a TOML config or the env. vars somewhere in `main`, and then pass some token context struct to the HTTP client.

But this can be refactored later.

Reply:
I agree with passing the token context to the HTTP client 👍
How would the toml look like?

I can work on this after this is merged btw 👍

Reply:
I was more wishing to have a better dependency flow in the app, to make it easier to understand how are the tokens acquired, and also to make it easier to test. Otherwise I'm fine with env. variables.

But if we went for TOML, I was thinking that it would look e.g. like this:
```toml
[tokens]
rust-lang = "foo"
rust-lang-nursery = "bar"
```
Then we would have a job matrix in CI, which would acquire the individual tokens and write them as outputs, and a follow-up job that would concatenate these tokens into the TOML file and pass it to `sync-team`.

Reply:
Thanks! I think it's better to discuss this in a separate issue after we merge this PR 👍

Reply:
Imo we should avoid writing tokens to a file, so I'd stick with reading them from the environment. Ideally as part of an `Configuration` struct (or similar) that acts as documentation as well. If environment variables are read in different places, it gets very difficult to understand which settings exist and which are required. (Speaking from experience working on other code bases. 🙈)